From 1e8bef8378d146727098f8e68b82e1b535f1e69d Mon Sep 17 00:00:00 2001
From: Zhao Liang <leo.zhao@intel.com>
Date: Mon, 23 Sep 2013 10:29:03 +0800
Subject: Enable video cross process buffer sharing

BZ: 147281

Setup service "media.IntelBufferSharing" for sharing peers

Category: Feature Differentiation
Domain: Video.Media-Framework
Origin: Internal
Upstream-Candidate: No

Change-Id: I217dc10cd7a37e0aacf93a887eb0397c3a1abbe3
Orig-Change-Id: Id787bc6f9892738b66523bf4c7cfe2a79d61857d
Signed-off-by: Zhao Liang <leo.zhao@intel.com>
Reviewed-on: http://android.intel.com:8080/133545
Reviewed-by: Yuan, Shengquan <shengquan.yuan@intel.com>
Tested-by: Post, DavidX J <davidx.j.post@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/mediaserver/Android.mk           |    6 ++++++
 media/mediaserver/main_mediaserver.cpp |    9 +++++++++
 2 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/media/mediaserver/Android.mk b/media/mediaserver/Android.mk
index 1ac647a..944cd3c 100644
--- a/media/mediaserver/Android.mk
+++ b/media/mediaserver/Android.mk
@@ -25,6 +25,10 @@ LOCAL_SHARED_LIBRARIES := \
 	liblog \
 	libbinder
 
+ifeq ($(INTEL_VIDEO_XPROC_SHARING),true)
+LOCAL_SHARED_LIBRARIES += libintelmetadatabuffer
+endif
+
 LOCAL_STATIC_LIBRARIES := \
 	libregistermsext
 
@@ -34,6 +38,8 @@ LOCAL_C_INCLUDES := \
     frameworks/av/services/audioflinger \
     frameworks/av/services/camera/libcameraservice
 
+LOCAL_C_INCLUDES += $(TARGET_OUT_HEADERS)/libmix_videoencoder
+
 LOCAL_MODULE:= mediaserver
 
 include $(BUILD_EXECUTABLE)
diff --git a/media/mediaserver/main_mediaserver.cpp b/media/mediaserver/main_mediaserver.cpp
index d5207d5..0121833 100644
--- a/media/mediaserver/main_mediaserver.cpp
+++ b/media/mediaserver/main_mediaserver.cpp
@@ -35,6 +35,10 @@
 #include "MediaPlayerService.h"
 #include "AudioPolicyService.h"
 
+#ifdef INTEL_VIDEO_XPROC_SHARING
+#include "IntelMetadataBuffer.h"
+#endif
+
 using namespace android;
 
 int main(int argc, char** argv)
@@ -128,6 +132,11 @@ int main(int argc, char** argv)
         MediaPlayerService::instantiate();
         CameraService::instantiate();
         AudioPolicyService::instantiate();
+
+#ifdef INTEL_VIDEO_XPROC_SHARING
+        IntelBufferSharingService::instantiate();
+#endif
+
         registerExtensions();
         ProcessState::self()->startThreadPool();
         IPCThreadState::self()->joinThreadPool();
-- 
1.7.4.1

